activestorage/CHANGELOG.md

```diff
@@ -1,3 +1,12 @@
+*   Support additional file checksum algorithms in S3 Adapter
+
+    Add support for CRC32, CRC32c, SHA1, SHA256, and CRC64NVMe to S3 service for
```

Comment:
I think that's where we might as misunderstood each others.

I don't think there is any value in supporting 6 different checksum algorithms for S3, the goal is just to change the current MD5 for something else (and to support both concurrently to allow smooth migration).

So we shouldn't add these 5, just one of these.

Reply:
I would choose SHA256 for my original intention of making things easier for FIPS compliance. This is a different choice than raw throughput/performance, so it seemed to make sense to support multiple for the different use cases. There is not much different in the adapter itself beyond MD5 vs non-MD5.

The other thing to perhaps take into account is that multipart uploads do not currently support checksums in the s3 adapter. If this was to be added, there are some differences based on the checksum algorithm (full file vs composite checksum) as noted under "Multipart uploads" section [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html)